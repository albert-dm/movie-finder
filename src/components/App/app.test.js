import React from 'react';
import { shallow } from 'enzyme';
import { App } from './index';

describe('Testing App behavior', () => {
  it('Should show loading', () => {
    const component = shallow(<App isLoading={true} error={''} />);
    expect(component).toMatchSnapshot();
  });
  it('Should show list if movies', () => {
    const component = shallow(<App isLoading={false} error={''} />);
    expect(component).toMatchSnapshot();
  });
  it('Should show error', () => {
    const component = shallow(<App isLoading={false} error={'Error message'} />);
    expect(component).toMatchSnapshot();
  });
});