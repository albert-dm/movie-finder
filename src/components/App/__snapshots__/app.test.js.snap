// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App behavior Should show error 1`] = `
<Fragment>
  <Toolbar
    actions={
      Array [
        <Memo(ConnectFunction) />,
        <Memo(ConnectFunction) />,
      ]
    }
    colored={true}
    component="header"
    prominent={true}
    singleColor={true}
    title="Movie Finder"
  />
  <div
    id="content"
  >
    <ErrorMessage
      message="Error message"
    />
  </div>
</Fragment>
`;

exports[`Testing App behavior Should show list if movies 1`] = `
<Fragment>
  <Toolbar
    actions={
      Array [
        <Memo(ConnectFunction) />,
        <Memo(ConnectFunction) />,
      ]
    }
    colored={true}
    component="header"
    prominent={true}
    singleColor={true}
    title="Movie Finder"
  />
  <div
    id="content"
  >
    <Connect(MovieList) />
  </div>
</Fragment>
`;

exports[`Testing App behavior Should show loading 1`] = `
<Fragment>
  <Toolbar
    actions={
      Array [
        <Memo(ConnectFunction) />,
        <Memo(ConnectFunction) />,
      ]
    }
    colored={true}
    component="header"
    prominent={true}
    singleColor={true}
    title="Movie Finder"
  />
  <div
    id="content"
  >
    <CircularProgress
      centered={true}
      determinateDashoffset={187}
      id="loading"
      scale={1}
      style={Object {}}
    />
  </div>
</Fragment>
`;
